deepfence_frontend/packages/ui-components/src/components/avatar/Avatar.tsx

```diff
@@ -32,28 +33,37 @@ const Child = ({ children }: { children: AvatarType['children'] }) => {
   );
 };
 
-export const Avatar = (props: AvatarType) => {
-  const {
-    asChild = false,
-    children = undefined,
-    src = '',
-    alt = '',
-    className = '',
-    onClick,
-  } = props;
-
-  return (
-    <button
-      onClick={onClick}
-      className={twMerge(
-        cx(
-          `inline-flex overflow-hidden relative justify-center items-center w-10 h-10 bg-gray-100 rounded-full dark:bg-gray-600`,
-          `text-gray-700 dark:text-gray-100 ${Typography.size.lg}`,
-        ),
-        className,
-      )}
-    >
-      {!asChild ? <img src={src} alt={alt} className="p-2" /> : <Child>{children}</Child>}
-    </button>
-  );
-};
+export const Avatar = forwardRef<HTMLButtonElement, AvatarType>(
+  (
+    {
+      asChild = false,
```

Comment:
why are we taking boolean `asChild` along with `children`. If children is `undefined` we can render icon, otherwise render `children`.

Reply:
Remove asChild props, if require more things we will update later.

Reply:
Remove asChild props, if require more things we will update later.